### PR TITLE
Use with-connection-pool rather than manual connection manager

### LINKED ISSUE #82: Handle NoHttpResponseException by using `clj-http.with-connection-pool`
NoHttpResponseExceptions result from Apache HTTP client connections that go stale i.e. when the server closes the connection, but the client keeps it open. The `clj-http` README offers this solution: https://github.com/dakrone/clj-http#nohttpresponseexception--due-to-stale-connections.

### PR DESCRIPTION
**Pull request by okal**
Closes #82

### CODE REVIEW

**Comment by pld** [comment]
Can you add some details on why this approach instead of the connection manager?

File: src/milia/api/io.clj [hunk 1]
```diff
@@ -139,11 +127,15 @@
   [method url http-options]
   (let [req-fn #(call-client-method method url (build-req http-options))]
     (try+  ; Catch all bad statuses
-     (try+ ; Catch 401 with token expire messages
-      (req-fn)
-      (catch #(expired-token? %) response
-        (do
-          (refresh-temp-token)
-          (req-fn))))
+     (client/with-connection-pool
```
Comment (pld) [inline src/milia/api/io.clj]: Can we put this in a separate function?

Comment (okal) [inline src/milia/api/io.clj]: Which part?

**Comment by okal** [comment]
Using `with-connection-pool` is the suggested solution to NoHttpResponseException in the clj-http README.

**Pull request merged**